CMakeLists.txt

```diff
@@ -562,23 +562,6 @@ if("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "LINUX")
     set(SWIFT_PRIMARY_VARIANT_ARCH_default "${SWIFT_HOST_VARIANT_ARCH}")
   endif()
 
-  # Compatible cross-compile SDKS for LINUX: ANDROID (arch always armv7)
-  is_sdk_requested(ANDROID swift_build_android)
-  if("${SWIFT_ANDROID_NDK_PATH}" STREQUAL "")
-    set(swift_can_crosscompile_stdlib_android FALSE)
-  else()
-    set(swift_can_crosscompile_stdlib_android TRUE)
-  endif()
-
-  if(swift_build_android AND ${swift_can_crosscompile_stdlib_android})
-    configure_sdk_unix(ANDROID "Android" "android" "android" "armv7" "armv7-none-linux-androideabi" "${SWIFT_ANDROID_SDK_PATH}")
-    # If we're not building for the host, the cross-compiled target should be the 'primary variant'.
-    if("${swift_build_linux}" STREQUAL "FALSE")
-      set(SWIFT_PRIMARY_VARIANT_SDK_default "ANDROID")
-      set(SWIFT_PRIMARY_VARIANT_ARCH_default "armv7")
```

Comment:
I mirrored what was going on with `configure_sdk_windows` below. I think the idea is that, right now, we limit what's built by specifying `TARGET_SDKS` and the like via CMake. So, in practice: no, we don't also build for macOS.

I do have a patch that addresses your concern, though, although I think you might find it "cheesy"! 😝 

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff085f6..af3cc70 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -713,8 +713,18 @@ if(swift_build_android AND NOT "${SWIFT_ANDROID_NDK_PATH}" STREQUAL "")
 
   if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
     set(_swift_android_prebuilt_suffix "darwin-x86_64")
+
+    if(swift_build_osx)
+      set(SWIFT_PRIMARY_VARIANT_SDK_default "ANDROID")
+      set(SWIFT_PRIMARY_VARIANT_ARCH_default "armv7")
+    endif()
   elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
     set(_swift_android_prebuilt_suffix "linux-x86_64")
+
+    if(swift_build_linux)
+      set(SWIFT_PRIMARY_VARIANT_SDK_default "ANDROID")
+      set(SWIFT_PRIMARY_VARIANT_ARCH_default "armv7")
+    endif()
   else()
     message(FATAL_ERROR "A Darwin or Linux host is required to build the Swift runtime for Android")
   endif()
```

The icky part is that `swift_build_osx` and `swift_build_linux` are defined several lines above, so it's not immediately clear where they come from. Thoughts? 🙌